docs/api-diffs/9.0/rc.1/diff.md

```diff
@@ -0,0 +1,1290 @@
+-            [DebuggerStepThroughAttribute]
+-            [ExperimentalAttribute("ASPIREEVENTING001", UrlFormat="https://aka.ms/dotnet/aspire/diagnostics#{0}")]
+-            public Task PublishAsync<T>(T @event, CancellationToken cancellationToken = default(CancellationToken)) where T : IDistributedApplicationEvent;
++            public Task PublishAsync<T>(T @event, CancellationToken cancellationToken = default(CancellationToken)) where T : IDistributedApplicationEvent;
+-            [ExperimentalAttribute("ASPIREEVENTING001", UrlFormat="https://aka.ms/dotnet/aspire/diagnostics#{0}")]
+-            public DistributedApplicationEventSubscription Subscribe<T>(IResource resource, Func<T, CancellationToken, Task> callback) where T : IDistributedApplicationResourceEvent;
++            public DistributedApplicationEventSubscription Subscribe<T>(IResource resource, Func<T, CancellationToken, Task> callback) where T : IDistributedApplicationResourceEvent;
+-            [ExperimentalAttribute("ASPIREEVENTING001", UrlFormat="https://aka.ms/dotnet/aspire/diagnostics#{0}")]
+-            public DistributedApplicationEventSubscription Subscribe<T>(Func<T, CancellationToken, Task> callback) where T : IDistributedApplicationEvent;
++            public DistributedApplicationEventSubscription Subscribe<T>(Func<T, CancellationToken, Task> callback) where T : IDistributedApplicationEvent;
+-            [ExperimentalAttribute("ASPIREEVENTING001", UrlFormat="https://aka.ms/dotnet/aspire/diagnostics#{0}")]
+-            public void Unsubscribe(DistributedApplicationEventSubscription subscription);
++            public void Unsubscribe(DistributedApplicationEventSubscription subscription);
+         }
+-        [ExperimentalAttribute("ASPIREEVENTING001", UrlFormat="https://aka.ms/dotnet/aspire/diagnostics#{0}")]
+-        public class DistributedApplicationEventSubscription
++        public class DistributedApplicationEventSubscription
+-        [ExperimentalAttribute("ASPIREEVENTING001", UrlFormat="https://aka.ms/dotnet/aspire/diagnostics#{0}")]
+-        public class DistributedApplicationResourceEventSubscription : DistributedApplicationEventSubscription
++        public class DistributedApplicationResourceEventSubscription : DistributedApplicationEventSubscription
++        public enum EventDispatchBehavior {
++            BlockingConcurrent = 1,
++            BlockingSequential = 0,
++            NonBlockingConcurrent = 3,
++            NonBlockingSequential = 2,
++        }
+-        [ExperimentalAttribute("ASPIREEVENTING001", UrlFormat="https://aka.ms/dotnet/aspire/diagnostics#{0}")]
+-        public interface IDistributedApplicationEvent
++        public interface IDistributedApplicationEvent
+         public interface IDistributedApplicationEventing {
++            Task PublishAsync<T>(T @event, EventDispatchBehavior dispatchBehavior, CancellationToken cancellationToken = default(CancellationToken)) where T : IDistributedApplicationEvent;
```

Comment:
This is a breaking change. While this was experimental, we may need to call this out in our release notes.

Reply:
I assume it would not break compilation and no work for UA?